### PR TITLE
add `pattern` param to checkout

### PR DESCRIPTION
**Pull request by nelsonni**

### CODE REVIEW

**Comment by wmhilton** [comment]
Thank you!!!

File: docs/checkout.md [hunk 1]
```diff
@@ -20,6 +23,13 @@ If the branch already exists it will check out that branch. Otherwise, it will c
...
+// checkout the master branch
 await git.checkout({ dir: '$input((/))', ref: '$input((master))' })
 console.log('done')
-```
\ No newline at end of file
+```
+
+```js live
+// checkout only JSON and Markdown files from master branch
+await git.checkout({ dir: '$input((/))', ref: '$input((master))', pattern: '*.{json,md}' })
```
Comment (wmhilton) [inline docs/checkout.md]:
```suggestion
await git.checkout({ dir: '$input((/))', ref: '$input((master))', pattern: '$input((**/*.{json,md}))' })
```

**Comment by wmhilton** [comment]
Whoops there's a little table-formatting error.
image

File: docs/checkout.md [hunk 2]
```diff
@@ -9,7 +9,10 @@ Checkout a branch
 | -------------------- | ------------------------ | -------------------------------------------------------------------------------------------------------------- |
 | fs [deprecated]      | FSModule                 | The filesystem containing the git repo. Overrides the fs provided by the [plugin system](./plugin_fs.md).      |
 | **dir**, gitdir      | string, string           | The [working tree](dir-vs-gitdir.md) directory path, and optionally the [git directory](dir-vs-gitdir.md) path |
-| **ref**              | string                   | Which branch to checkout.                                                                                      |
+| **ref**              | string                   | Which branch to checkout.
```
Comment (wmhilton) [inline docs/checkout.md]:
```suggestion
| **ref**              | string                   | Which branch to checkout.                                                                                      |
```

File: docs/checkout.md [hunk 3]
```diff
@@ -9,7 +9,10 @@ Checkout a branch
 | -------------------- | ------------------------ | -------------------------------------------------------------------------------------------------------------- |
 | fs [deprecated]      | FSModule                 | The filesystem containing the git repo. Overrides the fs provided by the [plugin system](./plugin_fs.md).      |
 | **dir**, gitdir      | string, string           | The [working tree](dir-vs-gitdir.md) directory path, and optionally the [git directory](dir-vs-gitdir.md) path |
-| **ref**              | string                   | Which branch to checkout.                                                                                      |
+| **ref**              | string                   | Which branch to checkout. 
+|
+| pattern              | string = null            | Filter the results to only those filepath matches a glob pattern.
```
Comment (wmhilton) [inline docs/checkout.md]:
```suggestion
| pattern              | string = null            | Filter the results to only those filepath matches a glob pattern.                                              |
```

File: docs/checkout.md [hunk 4]
```diff
@@ -9,7 +9,10 @@ Checkout a branch
 | -------------------- | ------------------------ | -------------------------------------------------------------------------------------------------------------- |
 | fs [deprecated]      | FSModule                 | The filesystem containing the git repo. Overrides the fs provided by the [plugin system](./plugin_fs.md).      |
 | **dir**, gitdir      | string, string           | The [working tree](dir-vs-gitdir.md) directory path, and optionally the [git directory](dir-vs-gitdir.md) path |
-| **ref**              | string                   | Which branch to checkout.                                                                                      |
+| **ref**              | string                   | Which branch to checkout. 
+|
```
Comment (wmhilton) [inline docs/checkout.md]:
```suggestion
```

File: docs/checkout.md [hunk 5]
```diff
@@ -9,7 +9,10 @@ Checkout a branch
...
+| pattern              | string = null            | Filter the results to only those filepath matches a glob pattern.
+|
```
Comment (wmhilton) [inline docs/checkout.md]:
```suggestion
```

**Comment by wmhilton** [comment]
whew, it'll be nice when Github finally lets us do multi-line suggestions. I _think_ I got it right?

**Comment by nelsonni** [comment]
@wmhilton I used your suggestions to fix the tables.

**Pull request merged**